dwave/plugins/torch/samplers/block_spin_sampler.py

```diff
@@ -0,0 +1,227 @@
+            its colour.
+        num_chains (int): Number of Markov chains to run in parallel.
+        proposal_acceptance_criteria (Literal["Gibbs", "Metropolis"]): The proposal acceptance
+            criterion used to accept or reject states in the Markov chain. Defaults to "Gibbs".
+        seed (Optional[int]): Random seed. Defaults to None.
+    """
+
+    def __init__(self, grbm: GRBM, crayon: Callable, num_chains: int,
+                 proposal_acceptance_criteria: Literal["Gibbs", "Metropolis"] = "Gibbs", seed=None):
+        super().__init__()
+        if num_chains < 1 or not isinstance(num_chains, int):
+            raise ValueError("Number of reads should be a positive integer.")
+        self._proposal_acceptance_criteria = proposal_acceptance_criteria.title()
+        if self._proposal_acceptance_criteria not in {"Gibbs", "Metropolis"}:
+            raise ValueError(
+                f'Proposal acceptance criterion should be one of "Gibbs" or "Metropolis"'
+            )
+        self._grbm = grbm
+        self._crayon = crayon
+        if not self._valid_crayon():
+            raise ValueError("`crayon` is not a valid colouring of `grbm`")
+        self._partition = self._get_partition()
+        self._padded_adjacencies, self._padded_adjacencies_weight = self._get_adjacencies()
+        self.rng = None
+        if seed is not None:
+            self.rng = torch.Generator()
+            self.rng.manual_seed(seed)
+        self.x = nn.Parameter(
+            rands((num_chains, grbm.n_nodes), generator=self.rng).float(),
+            requires_grad=False
+        )
+        self.zeros = nn.Parameter(torch.zeros((num_chains, 1)), requires_grad=False)
+
+    def _valid_crayon(self) -> bool:
+        """Determines whether `crayon` is a valid colouring of `grbm`.
+
+        Returns:
+            bool: True if the colouring is valid and False otherwise.
+        """
+        for u, v in self._grbm.edges:
+            if self._crayon(u) == self._crayon(v):
+                return False
+        return True
+
+    def _get_partition(self) -> nn.ParameterList:
+        """Computes the vertex partition induced by the colouring function `crayon`.
+
+        Returns:
+            nn.ParameterList: The partition induced by the colouring.
+        """
+        partition = defaultdict(list)
+        for node in self._grbm.nodes:
+            idx = self._grbm.node_to_idx[node]
+            c = self._crayon(node)
+            partition[c].append(idx)
+        partition = nn.ParameterList([
+            nn.Parameter(torch.tensor(partition[k], requires_grad=False), requires_grad=False)
+            for k in sorted(partition)
+        ])
+        return partition
+
+    def _get_adjacencies(self) -> tuple[list[torch.Tensor], list[torch.Tensor]]:
+        """Create two lists of padded adjacency lists (tensors), one for neighbouring indices and
+        another for the corresponding weight indices.
```

Comment:
I got a bit confused when reading this method. "edge indices" seems more intuitive for me than "weight indices", according to what is done in this method.

Reply:
It hasn't been that long since I wrote the code and I'm already confused by the documentation as well. Updated documentation.